h/models/group.py

```diff
@@ -121,6 +121,11 @@ def is_public(self):
     def __acl__(self):
         terms = []
 
+        # auth_clients that have the same authority as the target group
+        # may add members to it
+        add_member_principal = "client_authority:{}".format(self.authority)
+        terms.append((security.Allow, add_member_principal, 'member_add'))
```

Comment:
I think the local variable should probably be called `member_add_principal` to avoid mixing both `add_member` and `member_add` or, better, since `member_add` is actually the permission and the principal is `"client_authority:{}"`, call the variable `client_authority_principal`? But not a big deal